File: code_samples/typescript/testing_fixtures.ts
/**
 * Testing: Fixtures
 * AI/ML Training Sample
 */

interface IFixtures {
    data: string;
    process(input: string): void;
    validate(): boolean;
}

class Fixtures implements IFixtures {
    data: string;
    
    constructor() {
        this.data = "";
    }
    
    process(input: string): void {
        this.data = input;
    }
    
    getData(): string {
        return this.data;
    }
    
    validate(): boolean {
        return this.data.length > 0;
    }
}

// Example usage
const instance = new Fixtures();
instance.process("example");
console.log(`Data: ${instance.getData()}`);
console.log(`Valid: ${instance.validate()}`);

export { Fixtures, IFixtures };

File: code_samples/r/testing_fixtures.r
# Testing: Fixtures
# AI/ML Training Sample

Fixtures <- setRefClass(
    "Fixtures",
    fields = list(data = "character"),
    methods = list(
        initialize = function() {
            data <<- ""
        },
        process = function(input) {
            data <<- input
        },
        getData = function() {
            return(data)
        },
        validate = function() {
            return(nchar(data) > 0)
        }
    )
)

# Example usage
instance <- Fixtures$new()
instance$process("example")
cat("Data:", instance$getData(), "\n")
cat("Valid:", instance$validate(), "\n")
